lib/vm/src/func_data_registry.rs

```diff
@@ -0,0 +1,102 @@
+//! TODO: document this
+//!
+//! We store metadata for functions here. This is what a funcref actually points to.
```

Comment:
This seems a bit convoluted. Why do we need a registry of function metadata? Why we can't have the metadata on the definition itself?

Reply:
Generally speaking it's because of the distinction between local and imported functions and the case I mentioned above with arbitrary functions being allowed to be stored in Tables that the Wasm can interact with.

Additionally, the current model for `FuncRef`s is that they live as long as the engine / store does, so we store the data here.

I don't disagree that the current solution is more complicated than we want, but other than a few small improvements to the current system, I don't have a plan to fix it yet.